### ISSUE TITLE
Add conversion routines between `cartesian`, `xy` and `rc` conventions

### ISSUE THREAD

**Issue by soupault** [scikit-image/scikit-image#2275]
## Description

Unfortunately, there is no global agreement between developers of different image processing tools on the preferable coordinate convention yet. I'm aware of 3 most commonly used ones: `cartesian` (x - horiz. to the right, y - vert. to the top), `xy` (x - horiz. to the right, y - vert. to the bottom), `rc` (r - vert. to the bottom, c - horiz. to the right).
Even current `skimage` algorithms work with two of the above. Things become harder when people start to use several libraries in parallel (i.e. `skimage` + `OpenCV`).

I feel pretty sad reimplementing these routines over and over again for the new projects, and will be really happy to have them in the `skimage` distribution.

Above said, I propose to add to `skimage.util`: `cart_to_xy`, `cart_to_rc`, `xy_to_rc` and corresponding inverse functions, which will operate on list of coordinates and, possibly, images to convert points/image between these coordinate systems.

Please, let me know how you feel about this.

**Comment by stefanv** [comment]
Can you give an example of where you see this being used?

**Comment by soupault** [comment]
Here are some cases:

a. At the moment, I've some issues running ProjectiveTransform from `skimage`, so I'm using OpenCV for it. OpenCV works in `xy` coordinate system, so the typical call looks like this:
```
    points_fact = np.fliplr(points_fact)
    points_ref = np.fliplr(points_ref)
    coords_src = points_fact.astype(np.float32)[None, ...]
    coords_dst = points_ref.astype(np.float32)[None, ...]
    trans = cv2.getPerspectiveTransform(coords_src, coords_dst)
    ...
    points = np.fliplr(points)
    coords = points.astype(np.float32)[None, ...]
    points_out = cv2.perspectiveTransform(coords, trans)
    points_out = points_out[0, ...]
    points_out = np.fliplr(points_out)
```

b. Our own transforms use `xy` convention (see http://scikit-image.org/docs/dev/auto_examples/applications/plot_geometric.html#parameter-estimation). And, as long as https://github.com/scikit-image/scikit-image/issues/1789 is not fixed, this is an issue.

c. `cartesian` coordinate system is useful if user is willing to plot a _polygon_, created by one of our functions (is created in `rc`), __without__ an underlying image (if `imshow` is used, the yaxis is automatically flipped). In this case, they should either switch `xlims` explicitly (i.e. (800, 0)), or apply coordinate transformation on the points.

These are just quite simple cases, but I believe a lot of users got stumbled upon them.

**Comment by sciunto** [comment]
Quick comment. I'm always lost in these conventions and I tend to check each time I use a function I've never used before. I think that clarifications in the documentation (user guide) with schematics would be a nice addition too.

So +1.

**Comment by stefanv** [comment]
Are you saying both OpenCV and scikit-image use xy?

**Comment by soupault** [comment]
@stefanv that is certain.
OpenCV also does cause some other troubles with its Python bindings, but that is another story...

**Comment by jni** [comment]
@stefanv I've been using `transform` a lot more these past couple of months and can attest that the inconsistency between xy and rc has been frustrating! I think these functions would help.

In the long term, I'd consider writing a `transform_rc` module that uses numpy coordinates by convention, and would simply convert all coordinates transparently between the functions in `transform` and everything else. But perhaps this should be the subject of a forthcoming "SKIEP". =P

**Comment by stefanv** [comment]
I'm in favor of putting this one to bed by starting work on a transform module that makes the entire skimage consistent (wouldn't that be wonderful).

**Comment by alexandrejaguar** [comment]
@soupault I believe, as @stefanv said, that someday we could have a consistent set of tools (every function in `skimage` using `rc`, for instance). However, it would be nice to have this set of tools (`cart_to_xy` and so on) for compatibility with other DIP systems!

**Comment by soupault** [comment]
@alexandrejaguar exactly my thinking.
@stefanv why don't we have both? :) I've nothing against the proposed `transform_nd` module, but these tools would have another great application - compatibillity with OpenCV. Many of the people I know use both of the libraries (skimage & OpenCV) on a daily basis, and often - in a single pipeline.

**Comment by soupault** [comment]
OK, how about we forget these low-level coordinate convertion functions (https://github.com/scikit-image/scikit-image/pull/2278) and introduce OpenCV-compatibility tools instead (i.e. `img_to_opencv`, `coord_to_opencv`, same for `..._from_...`)? Would complement https://github.com/scikit-image/scikit-image/issues/2500 / https://github.com/scikit-image/scikit-image/pull/2519 nicely, I think.
Please, vote with :+1: / :-1: .

**Comment by alexandrejaguar** [comment]
@soupault I would be kindly against it. I think it would be better to have #2278 functions implemented and to use the documentation for presenting how to use them for converting OpenCV data.
I don't think it would be useful for us to have functions like `..._to_some_specific_toolkit`. You never know when another OpenCV is coming to town...
